File: hariom_sanpra/public/py/delivery_note.py
import frappe


def create_stock_entry(doc, method=None):
    if doc.custom_with_packing == 1 and doc.custom_packing_material:
        new_doc =  frappe.new_doc("Stock Entry")
        new_doc.stock_entry_type = "Material Issue"

        for row in doc.custom_packing_material:
            new_doc.append("items", {
            "s_warehouse": row.warehouse,
            "item_code": row.item_code,
            "qty":  row.qty
            })

        new_doc.save()

File: hariom_sanpra/public/js/delivery_note.js
frappe.ui.form.on('Packing Material', {
	item_code: function(frm, cdt, cdn) {
        get_stock(frm, cdt, cdn);
    },
    warehouse: function(frm, cdt, cdn) {
        get_stock(frm, cdt, cdn);
    }
})

function get_stock(frm, cdt, cdn){
    let row = locals[cdt][cdn];

    if (row.item_code && row.warehouse){
        frappe.call({
            method: "frappe.client.get_value",
            args: {
                doctype: "Bin",
                filters: {
                    item_code: row.item_code,
                    warehouse: row.warehouse
                },
                fieldname: ["actual_qty"]
            },
            callback: function(r){
                if (r.message){
                    frappe.model.set_value(cdt, cdn, "available_stock", r.message.actual_qty);
                } else {
                    frappe.model.set_value(cdt, cdn, "available_stock", 0);
                }
            }
        });
    }
}